scipy/interpolate/interpnd.pyx

```diff
@@ -113,6 +92,46 @@ class NDInterpolatorBase:
             self.scale = self.points.ptp(axis=0)
             self.scale[~(self.scale > 0)] = 1.0  # avoid division by 0
             self.points /= self.scale
+        
+        self._calculate_triangulation()
+        
+        if need_values or values is not None:
+            self._set_values(values)
+        else:
+            self.values = None
+        
+        self._xi = None
+        self._xi_simplices = None
+        self._xi_barycentric_coordinates = None
+
+    def _calculate_triangulation(self):
+        self.tri = qhull.Delaunay(self.points)
```

Comment:
`self.tri = self._calculate_triangulation(self.points)`.

Also maybe this can be added to LinearNDInterpolator and CTInterpolator only, and not to the Base. Yes, this breaks LSP a little but so what.

Reply:
Done.